drivers/gpio/gpio_rt1718s_port.c

```diff
@@ -0,0 +1,372 @@
+			ret = rt1718s_reg_read_byte(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						    &reg);
+			if (ret < 0) {
+				break;
+			}
+
+			if (value & BIT(pin)) {
+				new_reg = reg | RT1718S_REG_GPIO_CTRL_O;
+			} else {
+				new_reg = reg & ~RT1718S_REG_GPIO_CTRL_O;
+			}
+			ret = rt1718s_reg_update(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						 reg, new_reg);
+		}
+	}
+
+	k_sem_give(&data->lock);
+
+	return ret;
+}
+
+static int gpio_rt1718s_port_set_bits_raw(const struct device *dev, gpio_port_pins_t mask)
+{
+	const struct gpio_rt1718s_port_config *const config = dev->config;
+	struct gpio_rt1718s_port_data *const data = dev->data;
+	uint8_t new_reg, reg;
+	int ret = 0;
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	for (int pin = 0; pin < RT1718S_GPIO_NUM; pin++) {
+		if (mask & BIT(pin)) {
+			ret = rt1718s_reg_read_byte(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						    &reg);
+			if (ret < 0) {
+				break;
+			}
+			new_reg = reg | RT1718S_REG_GPIO_CTRL_O;
+			ret = rt1718s_reg_update(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						 reg, new_reg);
+		}
+	}
+
+	k_sem_give(&data->lock);
+
+	return ret;
+}
+
+static int gpio_rt1718s_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t mask)
+{
+	const struct gpio_rt1718s_port_config *const config = dev->config;
+	struct gpio_rt1718s_port_data *const data = dev->data;
+	uint8_t new_reg, reg;
+	int ret = 0;
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	for (int pin = 0; pin < RT1718S_GPIO_NUM; pin++) {
+		if (mask & BIT(pin)) {
+			ret = rt1718s_reg_read_byte(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						    &reg);
+			if (ret < 0) {
+				break;
+			}
+			new_reg = reg & ~RT1718S_REG_GPIO_CTRL_O;
+			ret = rt1718s_reg_update(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						 reg, new_reg);
+		}
+	}
+
+	k_sem_give(&data->lock);
+
+	return ret;
+}
+
+static int gpio_rt1718s_port_toggle_bits(const struct device *dev, gpio_port_pins_t mask)
+{
+	const struct gpio_rt1718s_port_config *const config = dev->config;
+	struct gpio_rt1718s_port_data *const data = dev->data;
+	uint8_t new_reg, reg;
+	int ret = 0;
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	for (int pin = 0; pin < RT1718S_GPIO_NUM; pin++) {
+		if (mask & BIT(pin)) {
+			ret = rt1718s_reg_read_byte(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						    &reg);
+			if (ret < 0) {
+				break;
+			}
+			new_reg = reg ^ RT1718S_REG_GPIO_CTRL_O;
+			ret = rt1718s_reg_update(config->rt1718s_dev, RT1718S_REG_GPIO_CTRL(pin),
+						 reg, new_reg);
+		}
+	}
+
+	k_sem_give(&data->lock);
+
+	return ret;
+}
+
+static int gpio_rt1718s_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+						enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_rt1718s_port_config *const config = dev->config;
+	struct gpio_rt1718s_port_data *const data = dev->data;
+	struct rt1718s_data *const data_rt1718s = config->rt1718s_dev->data;
+	uint8_t int8, mask8, new_mask8 = 0;
```

Comment:
`int8` being unsigned... confusing :-) These could use a better name.

Reply:
True. `int8` is the name of a register. It stands for interrupt.